stackslib/src/net/api/postblock_proposal.rs

```diff
@@ -0,0 +1,281 @@
+// Copyright (C) 2013-2020 Blockstack PBC, a public benefit corporation
+// Copyright (C) 2020-2023 Stacks Open Internet Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::io::{Read, Write};
+
+use clarity::vm::costs::ExecutionCost;
+use regex::{Captures, Regex};
+use stacks_common::codec::{
+    read_next, write_next, Error as CodecError, StacksMessageCodec, MAX_PAYLOAD_LEN,
+};
+use stacks_common::types::chainstate::{
+    BlockHeaderHash, BurnchainHeaderHash, ConsensusHash, StacksBlockId, StacksPublicKey,
+};
+use stacks_common::types::net::PeerHost;
+use stacks_common::types::StacksPublicKeyBuffer;
+use stacks_common::util::hash::{hex_bytes, Hash160, Sha256Sum};
+use stacks_common::util::retry::BoundReader;
+
+use crate::burnchains::affirmation::AffirmationMap;
+use crate::burnchains::Txid;
+use crate::chainstate::burn::db::sortdb::SortitionDB;
+use crate::chainstate::nakamoto::NakamotoBlock;
+use crate::chainstate::stacks::db::blocks::MINIMUM_TX_FEE_RATE_PER_BYTE;
+use crate::chainstate::stacks::db::StacksChainState;
+use crate::chainstate::stacks::{
+    StacksBlock, StacksBlockHeader, StacksTransaction, TransactionPayload,
+};
+use crate::core::mempool::MemPoolDB;
+use crate::cost_estimates::FeeRateEstimate;
+use crate::net::http::{
+    parse_json, Error, HttpBadRequest, HttpContentType, HttpNotFound, HttpRequest,
+    HttpRequestContents, HttpRequestPreamble, HttpResponse, HttpResponseContents,
+    HttpResponsePayload, HttpResponsePreamble, HttpServerError,
+};
+use crate::net::httpcore::{
+    request, HttpPreambleExtensions, HttpRequestContentsExtensions, RPCRequestHandler,
+    StacksHttpRequest, StacksHttpResponse,
+};
+use crate::net::p2p::PeerNetwork;
+use crate::net::relay::Relayer;
+use crate::net::{
+    Attachment, BlocksData, BlocksDatum, Error as NetError, StacksMessageType, StacksNodeState,
+};
+
+/// Represents a block proposed to the `v2/block_proposal` endpoint for validation
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct NakamotoBlockProposal {
+    /// Proposed block
+    pub block: NakamotoBlock,
+    /// Identify the stacks/burnchain fork we are on
+    pub parent_consensus_hash: ConsensusHash,
+    /// These are all the microblocks that the proposed block
```

Comment:
I don't think we're doing microblocks anymore

Reply:
Yeah, I removed that but forgot the comment